packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -75,26 +78,51 @@ const DRAG_OPTIONS: IDragOptions = {
   keepInBounds: true
 };
 
+// Sort participants in the following order.
+// 1. Video participants should always render before non-video participants.
+// 2. Video Tiles should be further sorted based on their ordering in dominant speakers list.
 const sortParticipants = (
-  participants: VideoGalleryRemoteParticipant[] | undefined
+  participants?: VideoGalleryRemoteParticipant[],
+  dominantSpeakers?: DominantSpeakers
 ): VideoGalleryRemoteParticipant[] => {
-  if (!participants) {
-    return [];
-  }
+  if (!participants) return [];
 
-  return participants.sort((p1, p2) => {
-    if (!p1?.videoStream?.renderElement?.childElementCount && !p2?.videoStream?.renderElement?.childElementCount) {
-      return 0;
-    }
-    if (!p1?.videoStream?.renderElement?.childElementCount) {
-      return 1;
-    }
-    if (!p2?.videoStream?.renderElement?.childElementCount) {
-      return -1;
-    }
+  const participantsWithVideo: VideoGalleryRemoteParticipant[] = [];
+  const participantsWithoutVideo: VideoGalleryRemoteParticipant[] = [];
 
-    return 0;
+  participants.forEach((p) => {
+    if (p.videoStream?.renderElement?.childElementCount) participantsWithVideo.push(p);
```

Comment:
Please don't use the short syntax for `if` that skips `{}`. It has caused severe security issues in the past.

If you don't mind walking the list twice, you can use a filter:

    const participantsWithVideo = participants.filter((p) => p.videoStream?.renderElement?.childElementCount > 0);
    const participantsWithoutVideo = participants.filter((p) => p.videoStream?.renderElement?.childElementCount === 0);

I do prefer your current use of `forEach`, but recommend adding curly braces.

Reply:
Sure, I can change it. But curious what kind of security issues it can cause.

Reply:
It's a very notorious SSL bug: https://www.imperialviolet.org/2014/02/22/applebug.html

Reply:
Haha. The way I used if/else would never have caused that. But added brackets anyway.